std/algorithm/searching.d

```diff
@@ -1063,6 +1063,29 @@ if (isBidirectionalRange!R1 &&
     }
 }
 
+/**
+   Checks whether the given
+   $(XREF_PACK_NAMED range,primitives,isInputRange,input range) ends with an
+   element that fulfils predicate $(D pred).
+
+   Params:
+
+   pred = Unary mandatory predicate.
+
+   doesThisEnd = The input range to check.
+
+   Returns:
+
+   $(D bool) if the first element in range $(D doesThisEnd) fulfils predicate
+   $(D pred), $(D false) otherwise.
+*/
+bool endsWith(alias pred = "a", R)(R doesThisEnd)
+    if (isInputRange!R &&
+        is(typeof(unaryFun!pred(doesThisEnd.front)) : bool))
```

Comment:
It's probably a good idea, though I don't know if changing the constraints for the other overloads would break any code. Regardless, I think your implementation of `isIfTestable` from the other PR would fit here.